xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java

```diff
@@ -430,19 +436,21 @@ private void saveFilterPreferences(EntityReference entityReference,
      * @return the result of the supplier
      * @throws E in case of error during the execution of the supplier
      */
-    private <T, E extends Throwable> T configureContextWrapper(SupplierErr<T, E> supplier) throws E
+    private <T, E extends Throwable> T configureContextWrapper(WikiReference wikiReference, SupplierErr<T, E> supplier)
+        throws E
     {
         XWikiContext context = this.contextProvider.get();
-        String currentWikiId = context.getWikiId();
+        WikiReference currentWiki = context.getWikiReference();
         if (this.filterPreferenceConfiguration.useMainStore()) {
             context.setWikiId(context.getMainXWiki());
+        } else if (wikiReference != null) {
+            context.setWikiReference(wikiReference);
         }
-
         try {
             return supplier.get();
         } finally {
-            if (!Objects.equals(currentWikiId, context.getWikiId())) {
-                context.setWikiId(currentWikiId);
+            if (wikiReference != null) {
```

Comment:
Not sure what is the point of this if, the context should get back to whatever was the context wiki and what's in this if is not related.

Reply:
You're right, I don't know why I wrongly assumed that if there was no wikiReference parameter there was no point to reset it. It's only true with the localStore...